src/cc_catalog_airflow/dags/provider_api_scripts/europeana.py

```diff
@@ -0,0 +1,282 @@
+            start_timestamp,
+            end_timestamp,
+            cursor
+        )
+
+        if next_cursor is None:
+            break
+
+        cursor = next_cursor
+
+        if image_list is not None:
+            images_stored = _process_image_list(image_list)
+            logger.info(
+                f'Images stored: {images_stored} of {total_number_of_images}')
+
+        else:
+            logger.warning('No image data!  Attempting to continue')
+
+    return images_stored
+
+
+def _get_image_list(
+        start_timestamp,
+        end_timestamp,
+        cursor,
+        endpoint=ENDPOINT,
+        max_tries=6  # one original try, plus 5 retries
+):
+    for try_number in range(max_tries):
+
+        query_param_dict = _build_query_param_dict(
+            start_timestamp,
+            end_timestamp,
+            cursor
+        )
+
+        response = delayed_requester.get(
+            endpoint,
+            params=query_param_dict,
+        )
+
+        logger.debug('response.status_code: {response.status_code}')
+        response_json = _extract_response_json(response)
+        image_list, next_cursor, total_number_of_images = _extract_image_list_from_json(
+            response_json)
+
+        if image_list is not None:
+            break
+
+    if try_number == max_tries - 1 and (
+            (image_list is None) or (next_cursor is None)):
+        logger.warning('No more tries remaining. Returning None types.')
+        return None, None, None
+    else:
+        return image_list, next_cursor, total_number_of_images
+
+
+def _extract_response_json(response):
+    if response is not None and response.status_code == 200:
+        try:
+            response_json = response.json()
+        except Exception as e:
+            logger.warning(f'Could not get image_data json.\n{e}')
+            response_json = None
+    else:
+        response_json = None
+
+    return response_json
+
+
+def _extract_image_list_from_json(response_json):
+    if (
+            response_json is None
+            or str(response_json.get('success')) != "True"
+    ):
+        image_list, next_cursor, total_number_of_images = None, None, None
+    else:
+        image_list = response_json.get('items')
+        next_cursor = response_json.get('nextCursor')
+        total_number_of_images = response_json.get('totalResults')
+
+    return image_list, next_cursor, total_number_of_images
+
+
+def _process_image_list(image_list):
+    prev_total = 0
+    for image_data in image_list:
+        total_images = _process_image_data(image_data)
+        if total_images is None:
+            total_images = prev_total
+        else:
+            prev_total = total_images
+
+    return total_images
+
+
+def _process_image_data(image_data):
+    logger.debug(f'Processing image data: {image_data}')
+    license_url = _get_license_url(image_data.get('rights'))
+    if license_url is None:
+        return None
```

Comment:
This isn't necessary, since the `image_store.add_item` method will already handle the case where there's no (valid) license URL.